lib/utils.js

```diff
@@ -0,0 +1,89 @@
+/*!
+ * body-parser
+ * Copyright(c) Express Contributors
+ * MIT Licensed
+ */
+
+'use strict'
+
+/**
+ * Module dependencies.
+ */
+
+var bytes = require('bytes')
+var contentType = require('content-type')
+var typeis = require('type-is')
+
+/**
+ * Module exports.
+ */
+
+module.exports = {
+  getCharset,
+  normalizeOptions
+}
+
+/**
+ * Get the charset of a request.
+ *
+ * @param {object} req
+ * @api private
+ */
+
+function getCharset (req) {
+  try {
+    return (contentType.parse(req).parameters.charset || '').toLowerCase()
+  } catch {
+    return undefined
+  }
+}
+
+/**
+ * Get the simple type checker.
+ *
+ * @param {string} type
+ * @return {function}
+ */
+
+function typeChecker (type) {
+  return function checkType (req) {
+    return Boolean(typeis(req, type))
+  }
+}
+
+/**
+ * Normalizes the common options for all parsers.
+ *
+ * @param {object} options options to normalize
+ * @param {string} defaultType default content type
+ * @returns {object}
+ */
+function normalizeOptions (options, defaultType) {
+  if (!defaultType || typeof defaultType !== 'string') {
```

Comment:
Lets keep this simpler for now and just validate that defaultType is defined
```suggestion
  if (!defaultType) {
```